src/components/markdown-renderer/markdown-it-configurator/BasicMarkdownItConfigurator.tsx

```diff
@@ -19,11 +19,77 @@ import { MarkdownItParserDebugger } from '../markdown-it-plugins/parser-debugger
 import { spoilerContainer } from '../markdown-it-plugins/spoiler-container'
 import { tasksLists } from '../markdown-it-plugins/tasks-lists'
 import { twitterEmojis } from '../markdown-it-plugins/twitter-emojis'
-import { MarkdownItConfigurator } from './MarkdownItConfigurator'
+import { RawNoteFrontmatter } from '../../editor-page/note-frontmatter/note-frontmatter'
+import { TocAst } from 'markdown-it-toc-done-right'
+import { LineMarkers, lineNumberMarker } from '../replace-components/linemarker/line-number-marker'
+import { plantumlWithError } from '../markdown-it-plugins/plantuml'
+import { headlineAnchors } from '../markdown-it-plugins/headline-anchors'
+import { KatexReplacer } from '../replace-components/katex/katex-replacer'
+import { YoutubeReplacer } from '../replace-components/youtube/youtube-replacer'
+import { VimeoReplacer } from '../replace-components/vimeo/vimeo-replacer'
+import { GistReplacer } from '../replace-components/gist/gist-replacer'
+import { legacyPdfShortCode } from '../regex-plugins/replace-legacy-pdf-short-code'
+import { legacySlideshareShortCode } from '../regex-plugins/replace-legacy-slideshare-short-code'
+import { legacySpeakerdeckShortCode } from '../regex-plugins/replace-legacy-speakerdeck-short-code'
+import { AsciinemaReplacer } from '../replace-components/asciinema/asciinema-replacer'
+import { highlightedCode } from '../markdown-it-plugins/highlighted-code'
+import { quoteExtraColor } from '../markdown-it-plugins/quote-extra-color'
+import { quoteExtra } from '../markdown-it-plugins/quote-extra'
+import { documentTableOfContents } from '../markdown-it-plugins/document-table-of-contents'
+import { frontmatterExtract } from '../markdown-it-plugins/frontmatter'
+
+export interface ConfiguratorDetails {
+  useFrontmatter: boolean,
+  onParseError: (error: boolean) => void,
+  onRawMetaChange: (rawMeta: RawNoteFrontmatter) => void,
+  onToc: (toc: TocAst) => void,
+  onLineMarkers?: (lineMarkers: LineMarkers[]) => void
+}
+
+export class BasicMarkdownItConfigurator<T extends ConfiguratorDetails> {
+  protected readonly options: T
+  protected configurations: MarkdownIt.PluginSimple[] = []
+  protected postConfigurations: MarkdownIt.PluginSimple[] = []
+
+  constructor(options: T) {
+    this.options = options
+  }
+
+  public pushConfig(plugin: MarkdownIt.PluginSimple): this {
+    this.configurations.push(plugin)
+    return this
+  }
+
+  public buildConfiguredMarkdownIt(): MarkdownIt {
+    const markdownIt = new MarkdownIt('default', {
+      html: true,
+      breaks: true,
```

Comment:
As I just see this: Maybe add a TODO here that this (`breaks`) needs to be updated according to the frontmatter later?

Reply:
I linked it to the redux state